src/main/java/xyz/parkh/doing/service/ScheduleService.java

```diff
@@ -0,0 +1,185 @@
+@RequiredArgsConstructor
+public class ScheduleService {
+
+    private final ScheduleMapper scheduleMapper;
+
+    // 접근 제한의 경우 부정을 먼저
+
+    // 일정 상세 조회
+    public ResponseEntity<ScheduleVo> findSchedule(final String userIdInJwt, final String userId, final Integer scheduleNo) {
+        // 필수 인자가 없는 경우
+        if (userIdInJwt == null || userId == null || scheduleNo == null) {
+            throw new ValueException("필수 인자가 없습니다.");
+        }
+
+        // TODO 친구 기능 추가 시 권한 확인하는 코드 작성 필요, 현재 본인 정보만 조회 가능.
+        // 본인이 아닐 경우
+        if (!userId.equals(userIdInJwt)) {
+            throw new DifferentAuthException("접근 할 수 있는 권한이 없습니다.");
+        }
+
+        ScheduleVo schedule = scheduleMapper.selectByNo(scheduleNo);
+
+        // 일정이 없을 경우
+        if (schedule == null) {
+            throw new ValueException("조회 할 일정이 없습니다.");
+        }
+
+        return ResponseEntity.ok().body(schedule);
+    }
+
+    // 목록 조회
+    // 1. 완료 O
+    // 2. 완료 X, 기한 O ( 기한 지난 것, 안 지난 것 구분)
+    // 3. 완료 X, 기한 X
+    public ResponseEntity<List<ScheduleShortInfo>> findScheduleList(
+            final String userIdInJwt, final String userId, final ScheduleStatusDto scheduleStatusDto) {
+        Boolean requestIsComplete = scheduleStatusDto.getIsComplete();
+        Boolean requestHasDeadLine = scheduleStatusDto.getHasDeadLine();
+
+        // 필수 인자가 없는 경우
+        if (userIdInJwt == null || userId == null || requestIsComplete == null) {
+            throw new ValueException("필수 인자가 없습니다.");
+        }
+
+        // 완료 되지 않은 일정 조회 시 기한이 있는 것 없는 것 구분 필수
+        if (!requestIsComplete && requestHasDeadLine == null) {
+            throw new ValueException("필수 인자가 없습니다.");
+        }
+
+        // TODO 친구 기능 추가 시 권한 확인하는 코드 작성 필요, 현재 본인 정보만 조회 가능.
+
+        // 본인이 아닐 경우
+        if (!userId.equals(userIdInJwt)) {
+            throw new DifferentAuthException("접근 할 수 있는 권한이 없습니다.");
+        }
+
+        List<ScheduleVo> scheduleList = scheduleMapper.selectByUserId(userId);
+
+        ArrayList<ScheduleShortInfo> scheduleShortInfoList = new ArrayList<>();
+
+        for (ScheduleVo schedule : scheduleList) {
+            Integer no = schedule.getNo();
+            String title = schedule.getTitle();
+            LocalDate endDate = schedule.getEndDate();
+            Boolean isComplete = schedule.getIsComplete();
+            LocalDate now = LocalDate.now();
+            Boolean overDeadLine = null;
+
+            // TODO 친구 기능 추가 시 조회한 schedule 의 isPublic 을 기반으로 공개 여부 추가 할 것.
+
+            // 완료된 일정 조회
+            if (requestIsComplete) {
+                if (!isComplete) continue; // 완료되지 않은 일정일 경우 처리하지 않음
+
+                // 기한이 있는 일정의 경우 기한 추가.
+                if (!Objects.isNull(endDate)) {
+                    overDeadLine = now.isAfter(endDate);
+                }
+
+                ScheduleShortInfo scheduleShortInfo = new ScheduleShortInfo().builder().no(no)
+                        .endDate(endDate).overDeadLine(overDeadLine).title(title).build();
```

Comment:
ScheduleShortInfo 객체 생성을 ScheduleVo 에 위임하면 어떨까요? 코드가 상당히 줄어들 것 같아요.
```suggestion
    schedule.toScheduleShortInfo()
    
    public ScheduleShortInfo toScheduleShortInfo() {
        return ScheduleShortInfo.Builder()
            .no(this.no)
            .endDate(this.endDate)
            .overDeadLine(LocalDate.now().isAfter(this.endDate))
            .title(this.title)
            .build();
    }
```